[ONNX] Deprecate setter functions for global variables

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85488
* #85487
* #85486
* #85350
* #85349
* #85312
* __->__ #85165
* #85491
* #84776
* #84728

`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

justinchuby committed:
[ONNX] Deprecate setter functions for global variables

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types

[ghstack-poisoned]

BowenBao commented on torch/_C/__init__.pyi.in:
```diff
@@ -404,7 +404,7 @@ def _jit_pass_onnx_block(
 ) -> Dict[Value, Value]: ...
 def _jit_pass_onnx_assign_scoped_names_for_node_and_value(graph: Graph) -> None: ...
-def _jit_pass_fixup_onnx_controlflow_node(n: Node, opset_version: _int) -> Node: ...
+def _jit_pass_fixup_onnx_controlflow_node(n: Node, opset_version: _int) -> List[Value]: ...
```
Please add this fix to description too.

justinchuby replied:
Done

BowenBao commented on torch/onnx/symbolic_helper.py:
```diff
@@ -1645,16 +1651,31 @@ def args_have_same_dtype(args):
 # TODO(justinchuby): Delete these setters, users should set the vars directly.
+@_deprecation.deprecated(
+    "1.13.0",
+    "TBD",
```
Why not set it to "1.14.0" as well?

justinchuby replied:
Sure thing!

BowenBao reviewed (approved):
LG, minor comments.

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`

[ghstack-poisoned]

AllenTiTaiWang commented on torch/onnx/_globals.py:
```diff
@@ -28,7 +27,9 @@ def __init__(self):
         # Whether the user's model is training during export
         self.export_training: bool = False
-        self.operator_export_type: Optional[_C_onnx.OperatorExportTypes] = None
+        self.operator_export_type: _C_onnx.OperatorExportTypes = (
```
I am planning on adding a new type (https://github.com/pytorch/pytorch/issues/84884). Any suggestion for that?

justinchuby replied:
Addressed offline

titaiwangms reviewed (approved):
LGTM!

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Fix return type of `_jit_pass_fixup_onnx_controlflow_node`
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Deprecate setter functions for global variables"


`_set_opset_version` and `_set_operator_export_type` are previously deprecated. This PR decorates them with the deprecation decorator, so warnings are emitted. 

- Remove usage of `_set_opset_version` and `_set_operator_export_type` in favor of setting the globals vars directly in torch.onnx internal
- Update `GLOBALS.operator_export_type`'s default to not be None to tighten types
- Remove usage of `_set_onnx_shape_inference`

[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge
